From 449f3e195ef1138f6ec868defffed9b80df224cc Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 031/296] Current view findFocus maybe null, which cause
 NullPointerException.

When run monkey test on TabHost application, easily reproduce the NullPointerException.
---
 core/java/android/widget/TabHost.java | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/core/java/android/widget/TabHost.java b/core/java/android/widget/TabHost.java
index 8696d0d5df7..4628f51481c 100644
--- a/core/java/android/widget/TabHost.java
+++ b/core/java/android/widget/TabHost.java
@@ -380,8 +380,11 @@ mTabHost.addTab(TAB_TAG_1, "Hello, world!", "Tab 1");
                     soundEffect = SoundEffectConstants.NAVIGATION_UP;
                     break;
             }
+
+            View focusView = mCurrentView.findFocus();
+
             if (event.getKeyCode() == keyCodeShouldChangeFocus
-                    && mCurrentView.findFocus().focusSearch(directionShouldChangeFocus) == null) {
+                    && (focusView == null || focusView.focusSearch(directionShouldChangeFocus) == null)) {
                 mTabWidget.getChildTabViewAt(mCurrentTab).requestFocus();
                 playSoundEffect(soundEffect);
                 return true;
-- 
2.11.0

